app/controllers/docs_controller.rb

```diff
@@ -7,23 +7,29 @@ class DocsController < ApplicationController
   def index; end
 
   def section
-    @doc = Document.find_by(section: @section, slug: "APEX")
+    @doc = Document.find_by!(section: @section, slug: "APEX")
     render action: :show
+  rescue ActiveRecord::RecordNotFound
+    render_404
   end
 
   def show
-    @doc = Document.where(section: @section).find(params[:slug])
+    @doc = @nav_docs.find(params[:slug])
```

Comment:
`@nav_docs` has the same code as `Document.where(section: @section)` and is set in `use_section`, which is automatically called.